gensim/matutils.py

```diff
@@ -600,13 +600,12 @@ def jaccard_distance(set1, set2):
 def dirichlet_expectation(alpha):
     """
     For a vector `theta~Dir(alpha)`, compute `E[log(theta)]`.
-
     """
     if len(alpha.shape) == 1:
         result = psi(alpha) - psi(np.sum(alpha))
     else:
         result = psi(alpha) - psi(np.sum(alpha, 1))[:, np.newaxis]
-    return result.astype(alpha.dtype)  # keep the same precision as input
+    return result
```

Comment:
Please return `astype`, because
`np.float32 -> np.float32`
`np.float64 -> np.float64`
but
`np.float16 -> np.float32`

Reply:
Oh, my bad, you're right!
Then tests that I added in separate file aren't needed.